File: backend/Mission10Assignment/Mission10Assignment/Data/Bowler.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Mission10Assignment.Data;

// Setting all the columns of the Bowlers table
public class Bowler
{
    [Key]
    public int BowlerID { get; set; }
    
    [Required]
    public string BowlerLastName { get; set; }
    [Required]
    public string BowlerFirstName { get; set; }

    public string? BowlerMiddleInit { get; set; }
    [Required]
    public string BowlerAddress { get; set; }
    [Required]
    public string BowlerCity { get; set; }
    [Required]
    public string BowlerState { get; set; }
    [Required]
    public string BowlerZip { get; set; }
    [Required] 
    public string BowlerPhoneNumber { get; set; }
    public int TeamID { get; set; }
    
    [ForeignKey("TeamID")]
    public Team? Team { get; set; }
    
}

File: frontend/src/types/bowler.ts
// This file defines the TypeScript type for a bowler, which includes their personal information and the team they belong to.

export type bowler = {
    bowlerID: number;
    bowlerLastName: string;
    bowlerFirstName: string;
    bowlerMiddleInit: string;
    bowlerAddress: string;
    bowlerCity: string;
    bowlerState: string;
    bowlerZip: number;
    bowlerPhoneNumber: string;
    teamID: number;
    team: {
        teamID: number;
        teamName: string;
    };
};